### DISCUSSION TITLE
SBIE cannot download to cryptomator vault

### DISCUSSION THREAD

**Discussion by fred919** [sandboxie-plus/Sandboxie#3398] [General]
I've been using SBIE for 15+ years. I've been donating somewhat randomly to the open source project since long before supporter certificates came along. I just switched from classic to plus recently to become "more mainstream" before asking for help with this issue.

My use case: I do all browsing in a sandbox. I have immediate recovery configured for the areas I use for downloads, so I must approve any downloads getting out of the sandbox. Most of my document folders are on OneDrive and encrypted by open source Cryptomator. If you're not familiar with it, Cryptomator stores everything encrypted and to use these files you request decryption, provide the password, and are presented with a virtual drive of the unencrypted versions. **When I try to download to a virtual drive mounted by Cryptomator**, it fails. The browser (Edge) reports "**Couldn't download - Download error**". I'm using:

- Windows 10 Pro with all available updates applied 
- MS Edge with latest updates (Version 118.0.2088.69 (Official build) (64-bit))
- Sandboxie-Plus Version: 1.11.4 (5.66.4)
- Cryptomator v1.10.1 (latest)

Prior to Cryptomator, I used Boxcryptor to do this for several years. They realized their exit strategy this year, sold to Dropbox, and the product is gone. The SBIE immediate recovery worked fine when downloading to the Boxcryptor virtual drive. I've fiddled with the Cryptomator options for mounting the virtual drive without any success. It's tiresome to download to a temp location, copy the file to the virtual drive, and shred the temp file.

Has anyone gotten SBIE to download to a Cryptomator virtual drive? Or can someone provide any specific troubleshooting/debug processes I could follow to gather more useful information about this? I'm a retired firewall/routing guy, not a developer, so be gentle. :-)

...Fred

**Comment by fred919** [comment]
The lack of traction on this surprises me. Did I post this in the wrong place? Cryptomator seems like a project that would be of interest to security-conscious folks like Sandboxie users. I avoided putting anything personal in the cloud until I discovered Cryptomator and Boxcryptor (RIP). Now I'm addicted to the convenience with the perceived safety. It's way easier than connecting to home with MFA VPN to access my NAS and everything sits encrypted until I want to access it. 

I'll probably get around to submitting a request though. I can't do the coding but I can certainly work with someone on the debug & test.

...Fred

**Reply by bastik-1001** [reply]
You can turn this discussion into an issue.

I don't know why it fails to save data in the virtual drive. To my understanding, that is surely something that should work.

About this not being addressed, lots of topics/discussions are not replied to by many people. Either they are not on Github or they have other reasons to "ignore" these discussions. The forums are not that alive as well. All of that could change, though.

**Comment by offhub** [comment]
Try different volume types.

Auto/WinFsp -> not working
WinFsp (Local Drive) -> OK

![sbiecryptomator01](https://github.com/sandboxie-plus/Sandboxie/assets/6871698/8d270063-5634-4f52-8d8e-4b3e08329ef2)

**Reply by fred919** [reply]
I've tried all the volume types on several occasions. I left it as WinFsp (Local Drive) even though it did not work because it seemed the best choice for this. I did get it to work tonight though, and here's how.

1. Enabled trace logging (I'm just getting to know Sandboxie-Plus) and noticed that the mountpoints are enumerated in the log but my virtual drive (P:) was not there. Of course not, as I only start cryptomator when I want to access the encrypted files.
2. Started cryptomator and mounted the virtual drive. Still didn't show in the trace log.
3. Reloaded the sbie config. The virtual drive mountpoint still didn't show in the log.
4. Terminated all processes in sbie, quit, and restarted sbie. Still didn't show in the log.
5. Tried downloading from sandboxed browser to virtual drive and it worked. So the drive has to be mounted (for me) when sbie starts for it to be available as download target. This is different (for me) than when I used boxcryptor's virtual drive.
6. Used 'wmic logicaldisk get name' at cmd prompt and surprise: the virtual drive does not show up. Nor would the 'dir' command show the virtual drive contents. That really surprises me since all the files show in file explorer. I need to try this more tomorrow.
7. Used the 'get-psdrive' command in powershell and it lists the virtual drive.

So I have a workaround but don't think I should have to use it. E.g., network drives mapped while sbie is already running should be available from sbie. So should this virtual drive, IMHO. In fact, I'll try cryptomator as a //UNC sometime just to see if sbie treats it any differently.

Thanks for the input,
Fred
